target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -59,6 +59,9 @@ ipq40xx_setup_interfaces()
 	compex,wpj428)
 		ucidef_set_interface_lan "eth0 eth1"
 		;;
+	engenius,emd1)
```

Comment:
Hm, this looks odd. From what I can deduce, you have a "unused" eth0 in your system, right?

Can't you just edit the dts and do

    soc {
          ....

                edma@c080000 {
                        status = "okay";
                        qcom,num_gmac = <1>;
                };
    };

    &gmac0 {
        vlan_tag = <1 0x20>;
    };

and add the "engenius,emd1" to the ```ucidef_set_interface_lan "eth0"``` list to get the same result with "eth0" now being the only ethernet adapter?

Reply:
Yes, you are right!
Thanks for your suggestion.
I've updated the codes.
Please have a review again.

Reply:
I have only one "eth0" now.

Reply:
I copied that from #2398 . The question is, if we should make this properly and also change the switchbitmaps and add the poll-link properties. Because from what I remember, your device  currently doesn't notify netifd when the cable gets unplugged and replugged/when the link gets downgraded.

I think the following would "work", but it needs testing (see the inline comment)

         soc {
                         ess-switch@c000000 {
                                 switch_lan_bmp = <0x20>;
                                 switch_wan_bmp = <0x00>; /* not sure without HW to test, if 0x00 works, if not I would accept 0x1e here */
                          };
         };

         &gmac0 {
                 local-mac-address = [00 00 00 00 00 00];
                 qcom,phy_mdio_addr = <4>;
                 qcom,poll_required = <1>;
                 qcom,forced_speed = <1000>;
                 qcom,forced_duplex = <1>;
                 vlan_tag = <1 0x20>;
         };

(In theory, you could also add a ```ucidef_add_switch "switch0" "0u@eth0" "5:lan"```, but this might end up just confusing people.)

Reply:
I've updated the "dts" file as you mentioned.
The ethernet works fine, and I can see kernel shows "eth0: GMAC Link is up with phy_speed=1000" and "eth0: GMAC Link is down" when I plug and unplug the cable.
But how do I check that it notifies netifd normally?

Reply:
You can check the carrier status of a device in netifd by using the ubus call network.device status

Reply:
Thanks to dedeckeh! The command is useful.
The device notifies netifd link status normally, please have a review again.